File: internal/content/testdata/bench/java/nc-eval-rpn.java
import java.util.*;
class Solution {
    public long eval_rpn(String[] tokens) {
        Deque<Long> stack = new ArrayDeque<>();
        Set<String> ops = new HashSet<>(Arrays.asList("+", "-", "*", "/"));
        for (String t : tokens) {
            if (ops.contains(t)) {
                long b = stack.pop(), a = stack.pop();
                if (t.equals("+")) stack.push(a + b);
                else if (t.equals("-")) stack.push(a - b);
                else if (t.equals("*")) stack.push(a * b);
                else stack.push((long)(a / b));
            } else {
                stack.push(Long.parseLong(t));
            }
        }
        return stack.pop();
    }
}

File: internal/content/testdata/bench/csharp/nc-eval-rpn.cs
using System.Collections.Generic;

public class Solution {
    public long eval_rpn(string[] tokens) {
        var stack = new Stack<long>();
        foreach (string t in tokens) {
            if (t == "+" || t == "-" || t == "*" || t == "/") {
                long b = stack.Pop(), a = stack.Pop();
                if (t == "+") stack.Push(a + b);
                else if (t == "-") stack.Push(a - b);
                else if (t == "*") stack.Push(a * b);
                else stack.Push((long)(a / b));
            } else {
                stack.Push(long.Parse(t));
            }
        }
        return stack.Pop();
    }
}
